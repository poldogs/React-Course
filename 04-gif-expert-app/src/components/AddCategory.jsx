import React, { useState } from 'react'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) => {
        e.preventDefault()
        setinputValue(e.target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault()
        if (inputValue.trim().length > 2) {
            setCategories(categories => [inputValue, ...categories])
            setinputValue('')
        }
    }


  return (
    <form onSubmit={ onSubmit }>
        <input 
        type="text" 
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={handleInputChange}/>
    </form>
    


  )
}
