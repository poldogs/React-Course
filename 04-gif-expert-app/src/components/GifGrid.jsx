import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs'

export const GifGrid = ({ category }) => {

  const [gifs, setGifs] = useState([])

  useEffect(() => {
    getGifs(category)
      .then(setGifs)
  }, [category])

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {
          gifs.map(({ id, title, url }) => (
            <div className="card" key={id}>
              <img src={url} alt={title} />
              <p>{title}</p>
            </div>
          ))
        }
      </div>
    </>
  )
}
