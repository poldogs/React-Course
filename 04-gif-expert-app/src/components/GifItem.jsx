import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({ id, title, url }) => (
    <li key={id} className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </li>
);

GifItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}