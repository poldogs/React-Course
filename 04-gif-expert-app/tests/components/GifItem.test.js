import React from 'react';
import { render } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('GifItem', () => {
    const title = 'Saitama';
    const url = 'https://onepunchman.fandom.com/wiki/Saitama';
    const id = '1234';

  test('matches the snapshot', () => {
    const { asFragment } = render(<GifItem {...{ title, url, id }} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('must show an img with src and alt indicated', () => {
    const { src, alt } = render(<GifItem {...{ title, url, id }} />).container.querySelector('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test('must show the title in the component', () => {
    render(<GifItem {...{ title, url, id }} />);
    expect(document.querySelector('p').textContent).toBe(title);
  });
  
});