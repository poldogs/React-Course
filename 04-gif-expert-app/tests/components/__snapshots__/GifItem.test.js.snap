// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem matches the snapshot 1`] = `
<DocumentFragment>
  <li
    class="card"
  >
    <img
      alt="Saitama"
      src="https://onepunchman.fandom.com/wiki/Saitama"
    />
    <p>
      Saitama
    </p>
  </li>
</DocumentFragment>
`;
